rhc-ose-ansible/roles/secure-registry/tasks/main.yaml

```diff
@@ -0,0 +1,164 @@
+
+  - name: "Add secret volumes to registry deploymentConfiguration"
+    command: "{{openshift.common.client_binary}} volume deploymentConfig/{{registry_dc}} --add --type=secret --secret-name={{registry_secret_name}} -m /etc/secrets --overwrite"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+    when: dc_cleanup.rc == 0
+    
+  - name: "Add secret volumes to registry deploymentConfiguration"
+    command: "{{openshift.common.client_binary}} volume deploymentConfig/{{registry_dc}} --add --type=secret --secret-name={{registry_secret_name}} -m /etc/secrets"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+    when: dc_cleanup.rc == 1
+
+  - name: "Enable TLS"
+    command: "{{openshift.common.client_binary}} env deploymentConfig/{{registry_dc}} REGISTRY_HTTP_TLS_CERTIFICATE={{registry_certificate}} REGISTRY_HTTP_TLS_KEY={{registry_key}} --overwrite"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Updating liveness probe to HTTPS scheme"
+    command: "{{openshift.common.client_binary}} patch deploymentConfig/{{registry_dc}} --api-version=v1 -p '{{liveness_patch}}'"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Updating readinessProbe to HTTPS scheme"
+    command: "{{openshift.common.client_binary}} patch deploymentConfig/{{registry_dc}} --api-version=v1 -p '{{readiness_patch}}'"
+    run_once: true
+    delegate_to: "{{ groups['masters'][0] }}"
+
+  - name: "Create registry certificates directory for serviceIP"
+    file: "state=directory path=/etc/docker/certs.d/{{registry_svc_ip}}:{{item}} mode=755"
+    with_items: "{{registry_svc_ports}}"
+
+  - name: "Create registry certificates directory for kubernetes service"
+    file: "state=directory path=/etc/docker/certs.d/{{registry_dc}}.default.svc.cluster.local:{{item}} mode=755"
+    with_items: "{{registry_svc_ports}}"
+
+  - name: "Copy registry cert to kubernetes service directory"
+    command: "scp {{registry_ca}} {{item[0]}}:/etc/docker/certs.d/{{registry_dc}}.default.svc.cluster.local:{{item[1]}}"
+    with_nested:
+      - "{{groups['nodes'][0]}}"
+      - "{{registry_svc_ports}}"
+    delegate_to: "{{groups['masters'][0]}}"
+
+  - name: "Copy registry cert to serviceIP directory"
+    command: "scp {{registry_ca}} {{item[0]}}:/etc/docker/certs.d/{{registry_svc_ip}}:{{item[1]}}"
+    with_nested:
+      -  "{{groups['nodes'][0]}}"
+      -  "{{registry_svc_ports}}"
+    delegate_to: "{{groups['masters'][0]}}"
+  
+  - name: "Remove --insecure flag from /etc/sysconfig/docker"
+    lineinfile: "dest=/etc/sysconfig/docker state=absent regexp='--insecure-registry=172.30.0.0/16'"
+
+  - name: "Remove INSECURE_REGISTRY flag from /etc/sysconfig/docker"
+    lineinfile: "dest=/etc/sysconfig/docker state=absent regexp='INSECURE_REGISTRYy=\'*172.30.0.0/16\'' "
+
+  - name: "reload daemon"
+    command: "systemctl daemon-reload"
+
+  - pause: seconds=10
+
+  - name: "restart docker"
```

Comment:
Firstly, why do we need to restart docker? Secondly, if we really need to do this, it should be a notify.